### PR TITLE
Fix prefetch C test for 32-bit

### PR DESCRIPTION
**Pull request by mppf**
All 64-bit x86 has prefetch, but some 32-bit intel did not
so it is necessary to supply -march=native for 32-bit testing.
This could be achieved working through the CHPL_TARGET_ARCH
mechanism to be more robust for different platforms, but I
believe that -march=native will work for all non-MIC architectures.

### CODE REVIEW

**Comment by mppf** [comment]
An alternative would be to skip this test for 32-bit

**Comment by mppf** [comment]
Or to set CHPL_TARGET_ARCH=native for 32-bit testing and then make the C tests include the SPECIALIZE flags.

**Comment by mppf** [comment]
Passed full local testing

**Comment by mppf** [comment]
@Kyle-B can you have a quick look at this change to prefetch.test.c ?

File: test/modules/standard/Prefetch/prefetch.compopts [hunk 1]
```diff
@@ -1 +1,5 @@
--O3
+# All 64-bit x86 has prefetch, but some 32-bit intel did not
+# so it is necessary to supply -march=native for 32-bit testing.
+# This could be achieved working through the CHPL_TARGET_ARCH
+# mechanism to be more robust for different platforms.
+-O3 -march=native
```
Comment (Kyle-B) [inline test/modules/standard/Prefetch/prefetch.compopts]: Setting `-march=native` is going to make this test fail on some of our runs (cce, and pgi). I'd rather just skipif 32-bit platforms than not run the test on those compilers.

**Comment by mppf** [comment]
I verified that the test is skipped on a 32-bit Linux system but not on my 64-bit mac.

**Pull request merged**